### ISSUE TITLE
[Suggestion] Smaller Addon footprint

### ISSUE THREAD

**Issue by ProPeach** [tsunamayo/Starship-EVO#3417]
Addons are a super cool way to buff and customise weapons, but their footprint is huge.
I suggest reducing their size so that a weapon core block with addons is still 1m/0.25m wide, because addons currently double a weapons width.
![image](https://user-images.githubusercontent.com/27857711/103182701-58b51c00-48a5-11eb-91cc-1860911c20ef.png)

This also has a balancing effect - A weapon core can have its DPS doubled with addons, but the addons size mean that the weapons footprint is also doubled, so the gross DPS per area remains the same. Currently, you're better off using two weapons next to eachother for max DPS rather than a fully modified single weapon. A smaller footprint is also easier to integrate into designs and makes turrets easier to build too.

**Comment by KINGrimson** [comment]
this would make addons a bigger part of the game if they where smaller then what  they are now as some avoid addons cause of their size or need to create weapons larger then they need to be due to the addons

**Comment by TIKIRobo** [comment]
it would be super nice to have these more inline addons so that our more exposed guns could look nicer

**Comment by ZachZent** [comment]
Yes please, they are rather bulky currently. And if I may add my own point, the damage addon is not symmetrical.
![image](https://user-images.githubusercontent.com/36980446/103183612-c561f380-4870-11eb-8c56-f6b1bb47ee86.png)

**Comment by YellowMiner** [comment]
I like being able to build a single barrel and use extra internal space to buff it up instead of adding more gun tho. And now i would just gain void to be filled with more hull, which potentially means smaller harder to target guns with more armor around them.

Or i could add an extra gun in this void if i can have an extra barrel sticking out, and the heat and power to do so, and i now have 2x the dps per volume using integrated addons.

(There was a whole weapon volume to ship size study done in starmade at one point which determined that increasingly smaller weapons mean more creative freedom (and potential DPS tho here it's balanced by heat and power), but also makes them harder to damage until it becomes pointless to attempt to shoot them and you're better to target another system.)

The point of addons is that they let you add power to the gun while expanding it in a different direction than just making the core longer or adding barrels. Using integrated addons would throw that out too.

What i'd suggest is to have addons come in an inline variant, working like barrels but stacking behind the core and disabling a core side slot each. Stats per volume would be maintained, you could have a fully upgraded weapon without extra width and make better use of your surface area, or just of your available space. More creative freedom without any significant stats change and solving your bulk issues.

(Then you could increase creative freedom further by having core variants (or checkboxes in the modules's window) with side slots in differents positions around the core (like a 4 tile long core with all slots arund one of them) but that's probably unnecessary complexity for most uses and outside teh scope of this issue)

**Comment by ZachZent** [comment]
As an extra idea for the models of modules, how about instead of a shape placed on top of a slot, it is a more rounded shape that when placed will complete the roundness of that section of the weapon. On top there can still be a shape which shows which module is placed
![image](https://user-images.githubusercontent.com/36980446/103240964-17595680-490f-11eb-8054-56ab92407f1c.png)

**Comment by nokturnihs** [comment]
How about add-ons being the "base" of the weapon - so for the small brick weapons, having roughly cube or whatever shaped weapons that scale like regular bricks? Want a six-addon chain gun? Go for it, It sits behind the barrel. Alternatively could convert add-ons to slottable chips and make a "weapon base" block that holds them. The barrel determines weapon type, range, accuracy, and the head determines what it usually does, ROF, recoil, etc.

**Comment by tsunamayo** [comment]
Ho another weapon addon debate. I am considering more radical solutions, people seems to not like them at all. I am waiting for the survival loop first to see how it could fit, I don't think driving every decision on look is right approach. 
Also I dont think gross DPS per area is the limiting factor, but I get it, people dont like the look of them.
Cheers
